tests/ballerina-test/src/test/java/org/ballerinalang/test/connectors/sql/SQLActionsTest.java

```diff
@@ -648,6 +654,416 @@ public void testLoadToMemorySelectAfterTableClose() throws Exception {
+          description = "Test mapping date to nillable int field")
+    public void testMappingDatesToNillableIntType() {
+        BValue[] returns = BRunUtil.invoke(result, "testMappingDatesToNillableIntType");
+        Assert.assertNotNull(returns);
+
+        long dateExpected = 1486060200000L;
+        long dateTimeExpected = 41745000L;
+        long timeExpected = 1486122780000L;
+        long timeStampExpected = 1486122780000L;
+
+        Assert.assertEquals(((BInteger) returns[0]).intValue(), dateExpected);
+        Assert.assertEquals(((BInteger) returns[1]).intValue(), dateTimeExpected);
+        Assert.assertEquals(((BInteger) returns[2]).intValue(), timeExpected);
+        Assert.assertEquals(((BInteger) returns[3]).intValue(), timeStampExpected);
+    }
+
+    @Test(groups = "ConnectorTest",
+          description = "Test mapping date to nillable string field")
+    public void testMappingDatesToNillableStringType() {
+        BValue[] returns = BRunUtil.invoke(result, "testMappingDatesToNillableStringType");
+        Assert.assertNotNull(returns);
+
+        String dateExpected = "2017-02-03+05:30";
+        String dateTimeExpected = "17:05:45.000+05:30";
+        String timeExpected = "2017-02-03T17:23:00.000+05:30";
+        String timeStampExpected = "2017-02-03T17:23:00.000+05:30";
+
+        Assert.assertEquals(returns[0].stringValue(), dateExpected);
+        Assert.assertEquals(returns[1].stringValue(), dateTimeExpected);
+        Assert.assertEquals(returns[2].stringValue(), timeExpected);
+        Assert.assertEquals(returns[3].stringValue(), timeStampExpected);
+    }
+
+    @Test(groups = "ConnectorTest",
+          description = "Test mapping to nillable type fields")
+    public void testMappingNullToNillableTypes() {
+        BValue[] returns = BRunUtil.invoke(result, "testMappingNullToNillableTypes");
+        Assert.assertNotNull(returns);
+        Assert.assertEquals(returns.length, 18);
+        Assert.assertNull(returns[0]);
+        Assert.assertNull(returns[1]);
+        ;
+        Assert.assertNull(returns[2]);
+        Assert.assertNull(returns[3]);
+        Assert.assertNull(returns[4]);
+        Assert.assertNull(returns[5]);
+        Assert.assertNull(returns[6]);
+        Assert.assertNull(returns[7]);
+        Assert.assertNull(returns[8]);
+        Assert.assertNull(returns[9]);
+        Assert.assertNull(returns[10]);
+        Assert.assertNull(returns[11]);
+        Assert.assertNull(returns[12]);
+        Assert.assertNull(returns[13]);
+        Assert.assertNull(returns[14]);
+        Assert.assertNull(returns[15]);
+        Assert.assertNull(returns[16]);
+        Assert.assertNull(returns[17]);
+    }
+
+    @Test(groups = "ConnectorTest",
+          description = "Test mapping nil to non-nillable int field",
+          expectedExceptions = BLangRuntimeException.class,
+          expectedExceptionsMessageRegExp = ".*" + TRYING_TO_ASSIGN_NIL_TO_NON_NILLABLE_FIELD + ".*")
```

Comment:
Instead of having concat at the beginning and end shall we define a constant with those two parts as well